File: bridge/java/TextMessage.java
package Bridge;

// UML diyagramındaki RefinedAbstraction kısmına denk gelmektedir.
// Abstraction (Message) sınıfından türer.
public class TextMessage extends Message {

    public TextMessage(IMessageSender messageSender) {
        super(messageSender);
    }

    @Override
    public void send() {
        messageSender.sendMessage(this);
    }
}

File: bridge/kotlin/TextMessage.kt
package Bridge

// UML diyagramındaki RefinedAbstraction kısmına denk gelmektedir.
// Abstraction (Message) sınıfından türer.
class TextMessage(messageSender: IMessageSender) : Message(messageSender) {
    override fun send() {
        messageSender.sendMessage(this)
    }
}
